packages/react-tabster/src/TabsterProvider.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import { makeMergePropsCompat, useMergedRefs } from '@fluentui/react-utilities';
+import { getCurrentTabster, createTabster, Types as TabsterTypes } from 'tabster';
+import { TabsterContext, TabsterContextValue } from './TabsterContext';
+
+export interface TabsterProvideProps extends React.HTMLAttributes<HTMLElement> {
+  dir?: 'rtl' | 'ltr';
+  document?: Document;
+  /**
+   * The root is automatically set as the `body` element of the ownerDocument.
+   * This prop needs to be set if a custom root is used
+   */
+  customRoot?: boolean;
+}
+
+export interface TabsterProviderState extends TabsterProvideProps {
+  dir: 'ltr' | 'rtl';
+  document?: Document;
+  contextValue: TabsterContextValue;
+}
+
+// eslint-disable-next-line deprecation/deprecation
+const mergeProps = makeMergePropsCompat<TabsterProviderState>();
+
+export const useTabsterProvider = (props: TabsterProvideProps, ref: React.Ref<HTMLElement>): TabsterProviderState => {
+  const rootRef = useMergedRefs(ref, React.useRef<HTMLElement>(null));
+  const state = mergeProps(
+    {
+      ref: rootRef,
+      as: 'div',
+      dir: 'ltr',
+    },
+    {},
+    props,
+  );
+
+  const tabsterOptions = { autoRoot: {} };
+  if (state.customRoot) {
+    delete tabsterOptions.autoRoot;
+  }
+
+  // only one instance per window of ability helpers should exist
+  let tabster: TabsterTypes.TabsterCore | undefined;
+  if (state.document?.defaultView) {
+    tabster =
+      getCurrentTabster(state.document.defaultView) ?? createTabster(state.document.defaultView, tabsterOptions);
+  }
+
+  // memoize context value so that it's stable
+  state.contextValue = React.useMemo<TabsterContextValue>(
+    () => ({ focusable: tabster?.focusable, tabsterInstance: tabster }),
```

Comment:
let's call the instance `tabster` since `ahInstance` was just to avoid `ah` :D

```suggestion
    () => ({ focusable: tabster?.focusable, tabster }),
```